internal/infra/powertrade/client.go

```diff
@@ -0,0 +1,251 @@
+package websocket
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/goemon/internal/domain/entities"
+	"github.com/gorilla/websocket"
+)
+
+type PowerTradeClient struct {
+	conn      *websocket.Conn
+	mu        sync.Mutex
+	writeMu   sync.Mutex
+	isRunning bool
+	done      chan struct{}
+	ticker    *time.Ticker
+}
```

Comment:
```suggestion
type PowerTradeClient struct {
	conn    *websocket.Conn
	mu      sync.Mutex
	writeMu sync.Mutex
	ctx     context.Context
	cancel  context.CancelFunc
	ticker  *time.Ticker
}
```
Use context for better cancellation